### ISSUE TITLE
Installation points to `libruby.so.2.7` on Linux

### ISSUE THREAD

**Issue by zedseven** [github/linguist#5808]
## Describe the bug
After a fresh install with `gem install github-linguist` in Arch Linux on a Ruby 3.0.3 installation, running the tool gives the following error:
```
$ github-linguist
<internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require': libruby.so.2.7: cannot open shared object file: No such file or directory - /home/zacc/.local/share/gem/ruby/3.0.0/gems/github-linguist-7.19.0/lib/linguist/linguist.so (LoadError)
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from /home/zacc/.local/share/gem/ruby/3.0.0/gems/github-linguist-7.19.0/lib/linguist/tokenizer.rb:2:in `<top (required)>'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from /home/zacc/.local/share/gem/ruby/3.0.0/gems/github-linguist-7.19.0/lib/linguist/classifier.rb:1:in `<top (required)>'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from /home/zacc/.local/share/gem/ruby/3.0.0/gems/github-linguist-7.19.0/lib/linguist/language.rb:9:in `<top (required)>'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from /home/zacc/.local/share/gem/ruby/3.0.0/gems/github-linguist-7.19.0/lib/linguist.rb:5:in `<top (required)>'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from /home/zacc/.local/share/gem/ruby/3.0.0/gems/github-linguist-7.19.0/bin/github-linguist:5:in `<top (required)>'
        from /home/zacc/.local/share/gem/ruby/3.0.0/bin/github-linguist:25:in `load'
        from /home/zacc/.local/share/gem/ruby/3.0.0/bin/github-linguist:25:in `<main>'
```
It seems like for some reason, it's pointing to libruby 2.7. I don't have a Ruby 2.7 installation on my machine, just a fresh install of Ruby 3.0.

This is the output of `ldd`:
```
$ ldd ~/.local/share/gem/ruby/3.0.0/gems/github-linguist-7.19.0/lib/linguist/linguist.so
        linux-vdso.so.1 (0x00007fff99344000)
        libruby.so.2.7 => not found
        libc.so.6 => /usr/lib/libc.so.6 (0x00007fa44da98000)
        /usr/lib64/ld-linux-x86-64.so.2 (0x00007fa44dccc000)
```

To fix it I tried installing Ruby 2.7, but the shared object files for Ruby 2.7 don't seem to even use the same naming format: (the 2.7 files are `libruby-<ver>.so` instead of `libruby.so.<ver>`)
```
$ exa -lag --group-directories-first --git /usr/lib | rg -i libruby
lrwxrwxrwx    20 root root 30 Nov  2021 libruby-2.7.so -> libruby-2.7.so.2.7.5
lrwxrwxrwx    20 root root 30 Nov  2021 libruby-2.7.so.2.7 -> libruby-2.7.so.2.7.5
.rwxr-xr-x  3.4M root root 30 Nov  2021 libruby-2.7.so.2.7.5
lrwxrwxrwx    16 root root 30 Nov  2021 libruby.so -> libruby.so.3.0.3
lrwxrwxrwx    16 root root 30 Nov  2021 libruby.so.3.0 -> libruby.so.3.0.3
.rwxr-xr-x  3.6M root root 30 Nov  2021 libruby.so.3.0.3
```
I'm not sure where this other format comes from.

The next thing I did is I tried creating a symbolic link from `libruby-2.7.so` to `libruby.so.2.7`. After doing this, Linguist could find the library but it complains it's an incompatible version:
```
$ github-linguist
<internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require': incompatible library version - /home/zacc/.local/share/gem/ruby/3.0.0/gems/github-linguist-7.19.0/lib/linguist/linguist.so (LoadError)
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from /home/zacc/.local/share/gem/ruby/3.0.0/gems/github-linguist-7.19.0/lib/linguist/tokenizer.rb:2:in `<top (required)>'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from /home/zacc/.local/share/gem/ruby/3.0.0/gems/github-linguist-7.19.0/lib/linguist/classifier.rb:1:in `<top (required)>'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from /home/zacc/.local/share/gem/ruby/3.0.0/gems/github-linguist-7.19.0/lib/linguist/language.rb:9:in `<top (required)>'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from /home/zacc/.local/share/gem/ruby/3.0.0/gems/github-linguist-7.19.0/lib/linguist.rb:5:in `<top (required)>'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from /home/zacc/.local/share/gem/ruby/3.0.0/gems/github-linguist-7.19.0/bin/github-linguist:5:in `<top (required)>'
        from /home/zacc/.local/share/gem/ruby/3.0.0/bin/github-linguist:25:in `load'
        from /home/zacc/.local/share/gem/ruby/3.0.0/bin/github-linguist:25:in `<main>'
```

Finally, I tried making a symbolic link from `libruby.so.3.0` to `libruby.so.2.7` (to trick it into using 3.0) and suddenly it works as expected. Here's my libruby file setup after doing this:
```
$ exa -lag --group-directories-first --git /usr/lib | rg -i libruby
lrwxrwxrwx    20 root root 30 Nov  2021 libruby-2.7.so -> libruby-2.7.so.2.7.5
lrwxrwxrwx    20 root root 30 Nov  2021 libruby-2.7.so.2.7 -> libruby-2.7.so.2.7.5
.rwxr-xr-x  3.4M root root 30 Nov  2021 libruby-2.7.so.2.7.5
lrwxrwxrwx    16 root root 30 Nov  2021 libruby.so -> libruby.so.3.0.3
lrwxrwxrwx    14 root root 26 Feb 04:11 libruby.so.2.7 -> libruby.so.3.0             // <-- here is the change I made
lrwxrwxrwx    16 root root 30 Nov  2021 libruby.so.3.0 -> libruby.so.3.0.3
.rwxr-xr-x  3.6M root root 30 Nov  2021 libruby.so.3.0.3
```

To summarise, it seems like for some reason Linguist is expecting Ruby 3.0 but looks for it under the name of Ruby 2.7. I literally installed Ruby so I could install this tool, so it's not like I have weird custom settings.

### Expected behaviour
I think I explained most of it already above, but I expected Linguist to work after installation and this library version issue was not expected.

### Additional notes
Output of `ruby --version`:
```
ruby 3.0.3p157 (2021-11-24 revision 3fb7d2cadc) [x86_64-linux]
```
Output of `gem --version`:
```
3.3.5
```
